src/LLMProviders/embeddingManager.ts

```
@@ -127,33 +126,29 @@ export default class EmbeddingManager {
   }
 
   getEmbeddingsAPI(): Embeddings | undefined {
-    const { embeddingModelKey } = getSettings();
+    const modelKey = getModelKey();
 
-    if (!EmbeddingManager.modelMap.hasOwnProperty(embeddingModelKey)) {
-      throw new CustomError(`No embedding model found for: ${embeddingModelKey}`);
+    if (!EmbeddingManager.modelMap.hasOwnProperty(modelKey)) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)